docgen/parser/parser.go

```diff
@@ -220,16 +220,39 @@ func parseMethods(pkgName string, messages map[string]*Message, methods []*googl
 			return nil, err
 		}
 
+		example, err := generateMethodExample(messages, m.GetOutputType())
+		if err != nil {
+			return nil, err
+		}
+
 		res[getQualifiedName(pkgName, m.GetName())] = &Method{
 			Name:      m.GetName(),
+			Request:   req,
 			Response:  messages[m.GetOutputType()],
+			Example:   example,
 			Operation: extOp.(*options.Operation),
-			Request:   req,
 		}
 	}
 	return res, nil
 }
 
+func generateMethodExample(messages map[string]*Message, name string) ([]string, error) {
+	_, ok := messages[name]
+	if !ok {
+		return nil, nil
+	}
+	p := genJSONExample(messages, name)
+	b, err := p.MarshalJSON()
```

Comment:
`MarshalJSON` already does [indentation](https://github.com/gunk/gunk/blob/636f76a60bcce64f4d27c0c9ee3c0c4a0a40f805/docgen/parser/parser.go#L371), but it's formatted specifically to look good in a [curl example](https://github.com/gunk/gunk/blob/995c81378bdeb318699e4cf08a191b2367a6519b/testdata/scripts/extract.txt#L570). i haven't looked into it too much but do you think maybe we could add properties (e.g. prefix, indent) to properties so that we don't do the indentation twice?

Reply:
> i haven't looked into it too much but do you think maybe we could add properties

we could, but I tried to remove the custom indent here and the generated markdown still looks good, so I think we should just remove it here. 

Removed and updated the test.

Reply:
removing the custom indentation affected the formatting though (the closing brackets aren't aligned). [example](https://github.com/gunk/gunk/blob/7c2723774205c59d4c49461de99b2df34c80327d/testdata/scripts/extract.txt#L569)

i think your initial approach is better from the end user's perspective. we indent twice, but at least the formatting is as expected

Reply:
I moved the indented codes out of MarshalJSON to avoid indenting twice.